sdk/python/foundation-models/system/inference/fill-mask/fill-mask-batch-endpoint.ipynb

```diff
@@ -0,0 +1,483 @@
+    "* We want this sample to run quickly, so save a smaller dataset containing a fraction of the original.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Define directories and filenames as variables\n",
+    "dataset_dir = \"book-corpus-dataset\"\n",
+    "training_datafile = \"train.jsonl\"\n",
+    "\n",
+    "batch_input_file = \"batch_input.csv\"\n",
+    "batch_dir = os.path.join(dataset_dir, \"batch\")\n",
+    "os.makedirs(batch_dir, exist_ok=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Load the ./book-corpus-dataset/train.jsonl file into a pandas dataframe and show the first 5 rows\n",
+    "pd.set_option('display.max_colwidth', 0) # set the max column width to 0 to display the full text\n",
+    "train_df = pd.read_json(os.path.join(\".\", dataset_dir, training_datafile), lines=True)\n",
+    "train_df.head()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Transform the data using the masking token."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Get the right mask token from huggingface\n",
+    "with urllib.request.urlopen(f\"https://huggingface.co/api/models/{model_name}\") as url:\n",
+    "    data = json.load(url)\n",
+    "    mask_token = data[\"mask_token\"]\n",
+    "\n",
+    "# Take the value of the \"text\" column, replace a random word with the mask token, and save the result in the \"masked_text\" column\n",
+    "train_df[\"masked_text\"] = train_df[\"text\"].apply(lambda x: x.replace(random.choice(x.split()), mask_token, 1))\n",
+    "\n",
+    "# Save the train_df dataframe to a jsonl file in the ./book-corpus-dataset folder with the `masked_` prefix\n",
+    "masked_datafile = os.path.join(\".\", dataset_dir, \"masked_\" + training_datafile)\n",
+    "train_df.to_json(masked_datafile, orient=\"records\", lines=True)\n",
+    "train_df.head()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Save a tenth of the input data to a file for testing batch inference. The MLflow model's signature specifies the input should be a column named `\"input_string\"`, so rename the transformed `\"masked_text\"` column. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "batch_df = train_df[['masked_text']].rename(columns={'masked_text': 'input_string'}).sample(frac=0.01)\n",
+    "batch_df.to_csv(os.path.join(batch_dir, batch_input_file), quoting=csv.QUOTE_ALL)\n",
+    "batch_df.head()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### 4. Deploy the model to a batch endpoint\n",
```

Comment:
Do we want to explain endpoints vs deployments here as well? Endpoint = Unified REST access point, Deployment = hosts model resources? Shouldn't be too much to add, but if this is going too much into Batch 101 territory for this sample, then maybe let's link to our overall batch explainer doc?

Reply:
add the what are batch endpoints link.